### PR TITLE
Implement the vendoring

### PR DESCRIPTION
**Pull request by mxgrey**
This implements the vendor package for `nlohmann_json_schema_validator`.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  find_package(nlohmann_json_schema_validator QUIET)
+endif()
+
+
+if(NOT nlohmann_json_schema_validator_FOUND)
+
+  set(setup_install_dir "${CMAKE_CURRENT_BINARY_DIR}/nlohmann_json_schema_validator")
+
+  include(ExternalProject)
+  ExternalProject_Add(nlohmann_json_schema_validator
```
Comment (arjo129) [inline CMakeLists.txt:21]: I remember @marcoag mentioning something about having issues with packaging libraries that need to be fetched from the internet using `FetchContent` and cousins. I might be completely wrong about this but I'll just tag him on this PR as this to be doubly sure as this is not my area of expertise.

File: package.xml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+<package format="2">
+  <name>nlohmann_json_schema_validator_vendor</name>
+  <version>0.1.0</version>
+  <description>A vendor package for JSON schema validator for JSON for Modern C++</description>
+  <maintainer email="grey@openrobotics.org">Grey</maintainer>
+  <license>Apache License 2.0</license>
+
+  <buildtool_depend>ament_cmake</buildtool_depend>
+
+  <depend>nlohmann_json_schema_validator</depend>
```
Comment (arjo129) [inline package.xml:12]: Don't we also need to depend on `nlohmann-json3`?

**Review by arjo129 — COMMENTED** [review]
I had some issues building this as a standalone package, I think we need to depend on `nlohmann-json3`. Once I manually installed `nlohmann-json3` exerything worked. I'm also not too sure `ExternalProject_Add` will play well with the packaging architecture (I may be completely wrong about this). Requesting that @marcoag takes a look at it once.

**Comment by mxgrey** [comment]
I'm not sure why GitHub isn't accepting @marcoag 's review. As far as I can tell the branch protection settings should be happy. Maybe it's because @arjo129 left a comment and hasn't given an approval since then?

**Review by arjo129 — APPROVED** [review]

**Comment by mxgrey** [comment]
@arjo129 Yep, you called it. I forgot to add the `open-robotics` team as maintainers. Seems to work now :ok_hand:

**Pull request merged**